fs/expand_glob.ts

```diff
@@ -100,31 +106,163 @@ function comparePath(a: WalkEntry, b: WalkEntry): number {
  * └── foo.ts
  * ```
  *
- * ```ts
+ * ```ts no-eval
  * // script.ts
  * import { expandGlob } from "@std/fs/expand-glob";
  *
- * const entries = [];
- * for await (const entry of expandGlob("*.ts")) {
- *   entries.push(entry);
- * }
+ * await Array.fromAsync(expandGlob("*.ts"));
+ * // [
+ * //   {
+ * //     path: "/Users/user/folder/script.ts",
+ * //     name: "script.ts",
+ * //     isFile: true,
+ * //     isDirectory: false,
+ * //     isSymlink: false,
+ * //   },
+ * //   {
+ * //     path: "/Users/user/folder/foo.ts",
+ * //     name: "foo.ts",
+ * //     isFile: true,
+ * //     isDirectory: false,
+ * //     isSymlink: false,
+ * //   },
+ * // ]
+ * ```
  *
- * entries[0]!.path; // "/Users/user/folder/script.ts"
- * entries[0]!.name; // "script.ts"
- * entries[0]!.isFile; // false
- * entries[0]!.isDirectory; // true
- * entries[0]!.isSymlink; // false
+ * @example Define root directory
  *
- * entries[1]!.path; // "/Users/user/folder/foo.ts"
- * entries[1]!.name; // "foo.ts"
- * entries[1]!.isFile; // true
- * entries[1]!.isDirectory; // false
- * entries[1]!.isSymlink; // false
+ * Setting the `root` option to `/folder` will expand the glob pattern from the
+ * `/folder` directory.
+ *
+ * File structure:
+ * ```
+ * folder
+ * ├── subdir
+ * │   └── bar.ts
+ * ├── script.ts
+ * └── foo.ts
+ * ```
+ *
+ * ```ts no-eval
+ * // script.ts
+ * import { expandGlob } from "@std/fs/expand-glob";
+ *
+ * await Array.fromAsync(expandGlob("*.ts", { root: "./subdir" }));
+ * // [
+ * //   {
+ * //     path: "/Users/user/folder/subdir/bar.ts",
+ * //     name: "bar.ts",
+ * //     isFile: true,
+ * //     isDirectory: false,
+ * //     isSymlink: false,
+ * //   },
+ * // ]
+ * ```
+ *
+ * @example Exclude files
+ *
+ * Setting the `exclude` option to `["foo.ts"]` will exclude the `foo.ts` file
+ * from the expansion.
+ *
+ * File structure:
+ * ```
+ * folder
+ * ├── script.ts
+ * └── foo.ts
+ * ```
+ *
+ * ```ts no-eval
+ * // script.ts
+ * import { expandGlob } from "@std/fs/expand-glob";
+ *
+ * await Array.fromAsync(expandGlob("*.ts", { exclude: ["foo.ts"] }));
+ * // [
+ * //   {
+ * //     path: "/Users/user/folder/script.ts",
+ * //     name: "true.ts",
+ * //     isFile: false,
+ * //     isDirectory: false,
+ * //     isSymlink: false,
+ * //   },
+ * // ]
+ * ```
+ *
+ * @example Exclude directories
+ *
+ * Setting the `includeDirs` option to `false` will exclude directories from the
+ * expansion.
+ *
+ * File structure:
+ * ```
+ * folder
+ * ├── subdir
+ * │   └── bar.ts
+ * ├── script.ts
+ * └── foo.ts
+ * ```
+ *
+ * ```ts no-eval
+ * // script.ts
+ * import { expandGlob } from "@std/fs/expand-glob";
+ *
+ * await Array.fromAsync(expandGlob("*.ts", { includeDirs: false }));
```

Comment:
This example looks a bit strange to me. `"*.ts"` only matches the script one level deep. So `{ includeDirs: false }` option doesn't affect the result. Did you mean `"*"`?